### ISSUE TITLE
Steam will only start With Terminal and if Opened with Start icon will break

### ISSUE THREAD

**Issue by TheShadowOfHassen** [ValveSoftware/steam-for-linux#8232]
Steam client built October  13 2021
Ubuntu 20.10
I have not opted for Steam  Beta
I have checked for updates

Steam will not open with the icon in the GUI

I had originally reported this error with the problem in error #8222  but was asked to do a separate issue report.

I  had upgraded to Ubuntu 21.10 and when error happene,d after I reinstalled my graphic drivers. Originally I  thought it was a bad upgrade, but when I reinstalled Ubuntu and steam I still had the problem.
Steam ran through the terminal once I installed it, but then didn’t when I ran it through the icon afterwards. It worked again through the terminal when I ran ‘sudo apt-get install libnvidia-gl-390:i386’ as some forums said, but when I restarted the computer again and ran it with the icon, it didn’t.
So next I ran it in the terminal and gave the exact same message as mxdpeep’s except the first line states that I am running ubuntu 21.10. I tried reinstalling steam but it did not work.
So now the only way I can get it to run is to delete the entire .steam folder, in the home directory, then run it through termninal. It will print out this:
(username the stand in for my name which is the stand in of the home directory )

Running Steam on ubuntu 21.10 64-bit
STEAM_RUNTIME is enabled automatically
Steam runtime environment up-to-date!
Steam client's requirements are satisfied
WARNING: Using default/fallback debugger launch
/home/username/.local/share/Steam/ubuntu12_32/steam
[2021-11-13 14:42:06] Startup - updater built Oct 13 2021 19:47:06
Installing breakpad exception handler for appid(steam)/version(1634158817)
[2021-11-13 14:42:06] Loading cached metrics from disk (/home/username/.local/share/Steam/package/steam_client_metrics.bin)
[2021-11-13 14:42:06] Using the following download hosts for Public, Realm steamglobal
[2021-11-13 14:42:06] 1. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-11-13 14:42:06] 2. https://cdn.akamai.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-11-13 14:42:06] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
Installing breakpad exception handler for appid(steam)/version(1634158817)
[2021-11-13 14:42:06] Verifying installation...
[2021-11-13 14:42:06] Verification complete
Loaded SDL version 2.0.17-6744061
Gtk-Message: 14:42:07.023: Failed to load module "gail"
Gtk-Message: 14:42:07.024: Failed to load module "atk-bridge"

(steam:66375): Gtk-WARNING **: 14:42:07.026: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru-dark/gtk-2.0/main.rc:775: error: unexpected identifier 'direction', expected character '}'

(steam:66375): Gtk-WARNING **: 14:42:07.028: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru-dark/gtk-2.0/hacks.rc:28: error: invalid string constant "normal_entry", expected valid string constant
Installing breakpad exception handler for appid(steam)/version(1634158817)
Runtime for steamwebhelper: defaulting to /home/username/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
CAppInfoCacheReadFromDiskThread took 25 milliseconds to initialize
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Xlib:  extension "NV-GLX" missing on display ":1".
MESA-INTEL: warning: Haswell Vulkan support is incomplete
Xlib:  extension "NV-GLX" missing on display ":1".
MESA-INTEL: warning: Haswell Vulkan support is incomplete 

It then gives me the login screen. Once I log into steam loads with all my games.  If I quit steam it will run again if I use the terminal. If I click the steam icon nothing will happen. If I run it with the terminal, it will give the same output as before when it did not work.  To get it to work again, I have to delete .steam folder in my home directory again and run it with my terminal.

If I delete the .steam folder again and run the steam with the icon in the start menu, it creates another .steam folder but steam will not work.

I tried the fix for error #8179 as kisak-valve said in error  #8222 but it did not work.

kisak-valve said also that this could be done by conflicting packages with the NVIDIA driver so i looked and despite me trying to install the libnvidia-gl-390:i386 package all my drivers are 418 like the NVIDIA website recommends for my graphics card.

In summary, if I delete the .steam folder in home and then only run steam through the terminal it will work, but if the desktop icon is hit, it will not until the .steam folder and home is deleted.

Thanks for the help!

**Comment by kisak-valve** [comment]
Hello @TheShadowOfHassen, what video card and driver version are you using? The `libnvidia-gl-390:i386` package is specifically the 32 bit userspace for the nVidia 390 series driver, if you're otherwise using a different nVidia driver series, then the 390 in the userspace package name should change to go along with it.

Since you can get Steam to start, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

Setting that aside, my expectation would be that Steam would either work or fall over if your issue was video driver related. Pondering this from a different direction, when you login the first time, are you checking the remember password option and if you are, does it make a difference in behavior if that box is not checked? If that has an effect, then there might be a dbus daemon issue similar to #3501.

**Comment by TheShadowOfHassen** [comment]
My drivers are NVIDIA-418. I think i wrote it in an edit of my first post, and yes i do have the libnvidia-gl-418:i386 package I made certain of it.

here is the gist:
[https://gist.github.com/TheShadowOfHassen/999774660a4ed3b76d5d95f4361a19e4](url)

I did check the password option once but when i deleted the .steam folder once and had to log in again i didn't. It did not make any difference except I had to log in again the next time I ran steam through terminal.

**Comment by kisak-valve** [comment]
Thanks, that at least tells us that when Steam is running, it is using your Intel Haswell chipset and if there is an nVidia GPU in the system the extended diagnostics tool thinks it's either incompatible with nvidia 418 or has no vulkan capabilities, which is odd.

`Xlib: extension "NV-GLX" missing on display ":1".` in your terminal spew makes me suspect that there's an Xorg.conf config leftover from before the distro upgrade which is making X try to run on the nvidia GPU, but then hitting some fallback path to render anything on the screen at all.

If that guess is correct, can you try to configure the system to run the entire X session on the Intel chipset and see if that has an effect?

**Comment by TheShadowOfHassen** [comment]
So my computer is one of those with two graphic cards, and when i opened NVIDIA x server settings it was set to using both NVIDIA and Intel. I set it to Intel, reset the computer and ran steam, it worked. So then i tried using only the NVIDIA card and that worked also.

If i would guess then it must be that steam can't handle the dual graphic mode?

**Comment by zpangwin** [comment]
I just started seeing a similar issue last night (launching steam from menu, nothing happened but it worked if launched from terminal). I was able to fix it on my own today just now. Figured I'd post my solution in case it works for someone else.

### System Info:

Fedora Cinnamon 34, with proprietary nvidia drivers from the RpmFusion repo. Kept up-to-date with `dnf-automatic`.

    $ inxi -SG
    System:
      Host: <redacted> Kernel: 5.15.12-100.fc34.x86_64 x86_64 bits: 64
      Desktop: Cinnamon 5.2.7 Distro: Fedora release 34 (Thirty Four)
    Graphics:
      Device-1: NVIDIA GM204 [GeForce GTX 970] driver: nvidia v: 495.46
      Display: x11 server: X.Org 1.20.14 driver: loaded: nvidia
      unloaded: fbdev,modesetting,nouveau,vesa resolution: 1920x1080~60Hz
      OpenGL: renderer: NVIDIA GeForce GTX 970/PCIe/SSE2 v: 4.6.0 NVIDIA 495.46

### Before

Starting yesterday (2022-Jan-10), when I tried to launch Steam from a pinned menu shortcut (e.g. Cinnamon menu has a right-click option to 'Add to panel' and I was using this). I assume this just uses the shortcut located at `/usr/share/applications/steam.desktop` which has `Exec=/usr/bin/steam %U`. Anyway, when I launched via this I would either briefly see a Steam progress bar which would disappear or see nothing at all. There was no system tray icon. But there was a process left behind so I killed it (using full commands here for clarify but I have them as aliases `pg` and `pk` on my system):

    $ pgrep --list-full --full --ignore-case
    174930 bash /home/fd/.local/share/Steam/steam.sh
    $ pkill --signal 9 --full --ignore-case steam

Since it appeared to be crapping out, I figured I'd run it from the terminal to see if I got any useful error messages I could search the web for... but to my surprise, it worked fine when I ran it from the terminal. Played a gaming session last night that way with no issues.

### Solution

It was still doing the same thing today so I decided to try fixing it. The first thing I tried was to just remove and reinstall steam. That by itself did not work

    $ sudo dnf remove -y steam
    $ sudo dnf install -y steam

I did rename my `~/.steam` and `~/.local/share/Steam` folders with a ".bak" before doing this and after I finally had things working I deleted the new folders and renamed my backups... but in hindsight I don't think that step was necessary. I'll leave that up to you.

Next I thought maybe my "pinned" shortcut was cached and somehow messed up... so I removed that. Then I made a copy of the menu shortcut on my desktop and tried running from that

    $ cp -a -t ~/Desktop /usr/share/applications/steam.desktop

At this point, I noticed that the shortcut icon wasn't showing up and realized that the shortcut did not have execute permissions. So I ran:

    $ chmod a+x ~/Desktop/steam.desktop

Immediately, the icon was displated on the desktop copy of the shortcut. I ran it, Steam came up and did not disappear. So I closed it and tried the one in my main ("Start") menu in Cinnamon. But it had the same before as last night where it displayed briefly then disappeared. I killed the hidden process again (pkill command above).

Then I went ahead and ran:

    $ sudo chmod a+x /usr/share/applications/steam.desktop

And when I tried again, everything was working fine. Closed steam again and re-pinned to my panel, restored my original `~/.steam` and `~/.local/Steam` folders and it started right up; didn't even have to sign back in.

I imagine there might be multiple causes that result in the same behavior, but I hope this helps someone.

**Comment by pktiuk** [comment]
Same issue with Pop! OS 20.04

**Comment by cement-head** [comment]
Same situation here, with Pop_OS; NVIDIA 510 driver.

**Comment by doh-123** [comment]
FYI, I ran into the same problem in Fedora on Cinnamon.  Steam basically launches, but never draws any windows.  a "killall -9 steam" will kill it off ok, and it will launch fine from command line.

**This fixed it for me though....**
In the Menu Editor, I clicked to "Edit Desktop File" and edited the PrefersNonDefaultGPU setting from true to false, and everything started working fine.

**Comment by TheShadowOfHassen** [comment]
I upgraded to Ubuntu 22.04 and it fixed the problem, sort of. The X server settings have disappeared from my system, so I can't change it from one card to the other. My system seems to be stuck on my Nvidia graphics card(by what I can see of system info in gnome settings). Steam loads fine though, as do all of my games, except for Hollow Knight which lags if not on my descrete card.

**Comment by IvanBorislavovDimitrov** [comment]
Managed to resolve it by opening "Nvidia X server settings"
Then: "PRIME Profiles" -> Select "NVIDIA On-Demand"

**Comment by danilopinotti** [comment]
> FYI, I ran into the same problem in Fedora on Cinnamon. Steam basically launches, but never draws any windows. a "killall -9 steam" will kill it off ok, and it will launch fine from command line.
> 
> **This fixed it for me though....** In the Menu Editor, I clicked to "Edit Desktop File" and edited the PrefersNonDefaultGPU setting from true to false, and everything started working fine.

It worked for me too. Just adding, you need to logout to take effect.

**Comment by Lloir** [comment]
Happens here on Arch / Kubuntu / Fedora on AMD drivers. Works when ran from CLI

**Comment by mattaw** [comment]
> Happens here on Arch / Kubuntu / Fedora on AMD drivers. Works when ran from CLI

Same problem here, both beta and normal work from a terminal, but not from the application link. I got a bunch of amdgpu errors in my logs (KSystemLog): 

```
Composite Threa[42954]: segfault at 0 ip 0000000000000000 sp 0000722d9edf86c8 error 14 in steamwebhelper[400000+10000] likely on CPU 5 (core 2, socket 0)
Code: Unable to access opcode bytes at 0xffffffffffffffd6.```
